evm-ds/src/cps_executor.rs

```diff
@@ -286,6 +313,41 @@ impl<'a> Handler for CpsExecutor<'a> {
         offset_len: U256,
     ) -> Capture<(ExitReason, Vec<u8>), Self::CallInterrupt> {
         if self.enable_cps {
+            if let Some(result) =
+                self.stack_executor
+                    .precompiles()
+                    .execute(&mut StackExecutorHandle {
+                        executor: &mut self.stack_executor,
+                        code_address,
+                        input: &input,
+                        gas_limit: target_gas,
+                        context: &context,
+                        is_static,
+                    })
+            {
+                return match result {
+                    Ok(PrecompileOutput {
+                        exit_status,
+                        output,
+                    }) => Capture::Exit((ExitReason::Succeed(exit_status), output)),
+                    Err(PrecompileFailure::Error { exit_status }) => {
+                        Capture::Exit((ExitReason::Error(exit_status), Vec::new()))
+                    }
+                    Err(PrecompileFailure::Revert {
+                        exit_status,
+                        output,
+                    }) => Capture::Exit((ExitReason::Revert(exit_status), output)),
+                    Err(PrecompileFailure::Fatal { exit_status }) => {
+                        self.stack_executor
+                            .state_mut()
+                            .metadata_mut()
+                            .gasometer_mut()
+                            .fail();
+                        Capture::Exit((ExitReason::Fatal(exit_status), Vec::new()))
+                    }
+                };
+            }
+
```

Comment:
GitHub doesn't allow reviewing the code not in the PR, but a comment about line 374 with
```
match result {
    Capture::Exit(s) => Capture::Exit(s)
    Capture::Trap(_) =>   ....
```
I don't understand why we cannot simply pass the result. If CPS is not enabled, why should there be any handling whatsoever, and not just pass `self.stack_executor.call()` directly?  

If it is necessary, please explain in the comments.

The same in `create` method.

Reply:
I had to do repacking here because StackExecutor returns StackExecutor::CreateInterrupt which is infallible whereas CpsExecutor has different Trap type for Create and Call.